Naev build failure

### Steps To Reproduce
Steps to reproduce the behavior:
0. add naev to configuration.nix environment.systemPackages = with pkgs; [ ... naev ...];
1. nixos-rebuild switch

### Build log
```
/nix/store/039g378vc3pc3dvi9dzdlrd0i4q93qwf-binutils-2.39/bin/ld: map.o: undefined reference to symbol 'exp@@GLIBC_2.29'
/nix/store/039g378vc3pc3dvi9dzdlrd0i4q93qwf-binutils-2.39/bin/ld: /nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:728: naev] Error 1
make[2]: *** [Makefile:891: all-recursive] Error 1
make[1]: *** [Makefile:630: all] Error 2
make: *** [Makefile:436: all-recursive] Error 1
error: builder for '/nix/store/67s7phg8hx85ixnz6vphxq3hkc88k0sy-naev-0.5.0.drv' failed with exit code 2
```

### Additional context
Only build.

### Notify maintainers
@siraben 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[root@BStation:~]# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.80, NixOS, 22.11 (Raccoon), 22.11.466.596a8e828c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -12250,6 +12250,12 @@
     githubId = 314564;
     name = "Ryan Lahfa";
   };
+  ralismark = {
+    email = "nixpkgs@ralismark.xyz";
+    github = "ralismark";
+    githubId = 13449732;
+    name = "Temmie";
+  };
   raphaelr = {
     email = "raphael-git@tapesoftware.net";
     matrix = "@raphi:tapesoftware.net";

--- a/pkgs/games/naev/default.nix
+++ b/pkgs/games/naev/default.nix
@@ -1,42 +1,83 @@
-{ lib, fetchurl, stdenv, SDL, openal, SDL_mixer, libxml2, pkg-config, libvorbis
-, libpng, libGLU, libGL, makeWrapper, zlib, freetype }:
+{ lib
+, SDL2
+, SDL2_image
+, enet
+, fetchFromGitHub
+, freetype
+, glpk
+, intltool
+, libpng
+, libunibreak
+, libvorbis
+, libwebp
+, libxml2
+, luajit
+, meson
+, ninja
+, openal
+, openblas
+, pcre2
+, physfs
+, pkg-config
+, python3
+, stdenv
+, suitesparse
+}:
 
-let
+stdenv.mkDerivation rec {
   pname = "naev";
-  version = "0.5.0";
-  name = "${pname}-${version}";
-in
-stdenv.mkDerivation {
-  inherit name;
+  version = "0.10.4";
 
-  srcData = fetchurl {
-    url = "mirror://sourceforge/naev/ndata-${version}";
-    sha256 = "0l05xxbbys3j5h6anvann2vylhp6hnxnzwpcaydaff8fpbbyi6r6";
+  src = fetchFromGitHub {
+    owner = "naev";
+    repo = "naev";
+    rev = "v${version}";
+    sha256 = "sha256-2cMRmKeoF6x5+95GoDgsoZG9QQo7qATrw/X+335l6FE=";
+    fetchSubmodules = true;
   };
 
-  src = fetchurl {
-    url = "mirror://sourceforge/naev/${name}.tar.bz2";
-    sha256 = "0gahi91lmpra0wvxsz49zwwb28q9w2v1s3y7r70252hq6v80kanb";
-  };
+  buildInputs = [
+    SDL2
+    SDL2_image
+    enet
+    freetype
+    glpk
+    intltool
+    libpng
+    libunibreak
+    libvorbis
+    libwebp
+    libxml2
+    luajit
+    openal
+    openblas
+    pcre2
+    physfs
+    suitesparse
+  ];
 
-  buildInputs = [ SDL SDL_mixer openal libxml2 libvorbis libpng libGLU libGL zlib freetype ];
+  nativeBuildInputs = [
+    (python3.withPackages (ps: with ps; [ pyyaml mutagen ]))
+    meson
+    ninja
+    pkg-config
+  ];
 
-  nativeBuildInputs = [ pkg-config makeWrapper ];
+  mesonFlags = [
+    "-Ddocs_c=disabled"
+    "-Ddocs_lua=disabled"
+    "-Dluajit=enabled"
+  ];
 
-  env.NIX_CFLAGS_COMPILE = "-include ${zlib.dev}/include/zlib.h";
-
-  postInstall = ''
-    mkdir -p $out/share/naev
-    cp -v $srcData $out/share/naev/ndata
-    wrapProgram $out/bin/naev --add-flags $out/share/naev/ndata
+  postPatch = ''
+    patchShebangs --build dat/outfits/bioship/generate.py utils/build/*.py utils/*.py
   '';
 
   meta = {
     description = "2D action/rpg space game";
     homepage = "http://www.naev.org";
     license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [viric];
+    maintainers = with lib.maintainers; [ viric ];
     platforms = lib.platforms.linux;
-    hydraPlatforms = [];
   };
 }

--- a/pkgs/games/naev/default.nix
+++ b/pkgs/games/naev/default.nix
@@ -77,7 +77,7 @@ stdenv.mkDerivation rec {
     description = "2D action/rpg space game";
     homepage = "http://www.naev.org";
     license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [ viric ];
+    maintainers = with lib.maintainers; [ ralismark ];
     platforms = lib.platforms.linux;
   };
 }

